finagle-core/src/main/scala/com/twitter/finagle/NameTreeParsers.scala

```diff
@@ -55,8 +55,17 @@ private class NameTreeParsers private (str: String) {
   }
 
   private[this] def eatWhitespace() {
-    while (!atEnd && str(idx).isWhitespace)
+    while (!atEnd && (str(idx).isWhitespace || str(idx) == '#')) {
+      if (str(idx) == '#') eatLine()
```

Comment:
i saw the test case after i commented.

seems reasonable to me, but i'll defer to others working more closely on dtabs.